kafka/src/main/java/com/opentable/logging/KafkaAppender.java

```diff
@@ -79,7 +83,16 @@ public void stop()
     @Override
     protected void append(ILoggingEvent eventObject)
     {
-        producer.send(new ProducerRecord<>(topic, keyGenerator.next(), encoder.encode(eventObject)));
+        try {
+            producer.send(new ProducerRecord<>(topic, keyGenerator.next(), encoder.encode(eventObject)));
+        } catch (InterruptException e) {
+            if (Thread.interrupted()) {
```

Comment:
Dumbb question, when would this NOT be the case? So I think you are just using this to clear?